aiomonitor/monitor.py

```diff
@@ -154,85 +232,98 @@ def _filter_cmds(self, *,
                     if altname.startswith(startswith):
                         yield CmdName(altname, name)
 
-    def _command_dispatch(self,
-                          sin: IO[str],
-                          sout: IO[str],
-                          user_input: str) -> None:
-        comm, *args = user_input.split(' ')
-        cmd = self._getcmd(comm)
-        try:
-            cmd(sin, sout, *args)
-        except Exception as e:
-            msg = 'Exception occured during command "{}": {}'
-            msg = msg.format(user_input, repr(e))
-            sout.write(msg + '\n')
-            log.exception(msg)
-
-    def _getcmd(self, comm: str) -> Callable:
+    def map_args(self, cmd: Callable, args: Sequence[str]
+                 ) -> Generator[Any, None, None]:
+        params = inspect.signature(cmd).parameters.values()
+        ia = iter(args)
+        for param in params:
+            if (param.annotation is param.empty or
+                    not callable(param.annotation)):
+                type_ = lambda x: x  # noqa
+            else:
+                type_ = param.annotation
+            try:
+                if str(param).startswith('*'):
+                    yield from (type_(arg) for arg in ia)  # type: ignore
+                else:
+                    yield type_(next(ia))  # type: ignore
+            except StopIteration:
```

Comment:
Lets use `contextlib.suppress` instead `except StopIteration: pass`, looks a bit nicer

Reply:
Oh yes, How could I have overlooked the `suppress` for this case :)